checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsAnnotatedTypeFactory.java

```diff
@@ -430,4 +443,75 @@ private AnnotationMirror ensuresCMAnno(String[] expressions, List<String> called
     AnnotationMirror am = builder.build();
     return am;
   }
+
+  // TODO: Probably move (parts of) the next two methods up into AnnotatedTypeFactory, eventually.
```

Comment:
I don't see anything in the implementations of these methods that is specific to the Called Methods checker. Why not move these to `AnnotatedTypeFactory` now, so that this fix applies to all checkers? (The same issue can occur for other checkers, but it just occurs much less frequently, since the Called Methods hierarchy is both so large and so amenable to inference.)

Reply:
The only exception to this is the use of the `emptyCalledMethods` field in `makeConditionConsistentWithOtherMethod`. That can be replaced with the top annotation from the relevant hierarchy.

Getting the top AnnotationMirror in AnnotatedTypeFactory might be a little tricky, but in the worst case we can at least move this to `AccumulationAnnotatedTypeFactory` rather than restricting it to `CalledMethods`.